### ISSUE TITLE
Enable Roundcube Enigma plugin for PGP support

### ISSUE THREAD

**Issue by brycefisher** [mail-in-a-box/mailinabox#892]
Looking at the release of Roundcube used by MIAB, it has the Enigma plugin builtin by default. This plugin is designed to provide basic PGP to Roundcube. I'd love to contribute a PR which would [enable that plugin](https://github.com/mail-in-a-box/mailinabox/blob/master/setup/webmail.sh#L104) and basically just keeping on a branch until GPG worked as part of the setup script. Would such a PR be accepted?

**Comment by yodax** [comment]
Roundcube 1.2 has (limited) support for PGP. That has been merged but not released (in miab). Does enigma add anything additional?

**Comment by brycefisher** [comment]
Without Enigma enabled in Roundcube's plugin list, its impossible to use PGP as far as I can tell. 

I'm a new Roundcube user and MIAB user as well, but a quick [look through the Roundcube codebase on github](https://github.com/roundcube/roundcubemail/search?p=1&q=PGP&utf8=%E2%9C%93) shows that nearly all the references to PGP appear to be in the enigma plugin.

When I enabled the Enigma plugin in my MIAB install, two new preference panes appeared one for s/mime certificates and one for PGP key management.

After a little more googling about Enigma, I also found this article which suggests using the Mailvelope plugin instead of Enigma for full end-to-end encryption (keeping the PGP keys in the browser). I'm not sure about the quality of Mailvelope but that approach sounds interesting to me.

My ultimate goal would be to submit (or solicit) a PR which provided PGP support in Roundcube in MIAB.

**Comment by ChiefGyk** [comment]
@brycefisher I have played with Mailvelope a little bit with gmail a few years ago and it was nice. It worked rather well. I would have to see if it works for MiaB, as I have been using Mozilla Thunderbird + Enigmail. But I would rather Mailvelope than incorporate in MiaB. I just feel uneasy about my private keys publicly accessed on a server.
But it would still be a nice option to have for some offices I know to transmit info a bit more securely.

**Comment by brycefisher** [comment]
@ChiefGyk so, just to clarify, you would prefer Mailvelope being installed and enabled by default in MIAB?

**Comment by ChiefGyk** [comment]
@brycefisher yes Mailvelope ( mailvelope.com ) I think is a better solution as it stores it locally in a browser extension for Firefox and Chrome. Because with Roundcube if not mistaken that means uploading your private GnuPG key correct? I'm just thinking along the lines of worst case and someone gets into the server, they now have my private GnuPG key as well.

**Comment by brycefisher** [comment]
Totally agree -- thanks for clarifying! I'll try to work on a PR to that effect unless someone shouts at me :-)

**Comment by JoshData** [comment]
I'll accept a PR for enabling Roundcube's enigma plugin if it's well tested.

**Comment by brycefisher** [comment]
Awesome! I'm finally about to have some time to work on this. Thanks, @JoshData

**Comment by viogq** [comment]
enigma cannot find private key here, when it really should.
